charm_versions.md

```diff
@@ -0,0 +1,138 @@
+than one of an admin.
+
+Authors publish new versions of charm libraries using the
+[charmcraft publish-lib command](https://juju.is/docs/sdk/create-and-publish-a-charm-library#heading--update-a-charm-library),
+setting the [metadata](https://juju.is/docs/sdk/library#heading--structure) appropriately - in
+particular, the `LIBAPI` value for the API version (which is also in the path) and the `LIBPATCH`
+value for the patch version.
+
+To upgrade the revision of a lib used by a charm, the same `charmcraft fetch-lib` command is used
+as when originally installing the lib. To upgrade the version of the lib, use `fetch-lib` but
+change the version number in the path, for example from `charms.demo.v0.demo` to
+`charms.demo.v1.demo`. See the
+[documentation](https://juju.is/docs/sdk/find-and-use-a-charm-library#heading--update-a-charm-library)
+for more details.
+
+### Interfaces
+
+When integrating two charms together, it's important that both charms understand what the other
+side of the relation requires and provides. In the charm metadata this is declared using the
+[interface field of the relation](https://juju.is/docs/sdk/charmcraft-yaml#heading--peers-provides-requires).
+
+Interfaces are more meaningful when they are
+[registered](https://juju.is/docs/sdk/register-an-interface), which involves defining the data
+structures that each side of the relation will use in the relation databag, and creating a suite of
+tests that ensure that charms that require or provide the interface align with the specification.
+
+Interface versions are defined through the folder structure in the
+[charm-relation-interfaces repository](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/)
+and are updated by authors using pull requests.
+
+## Workload / Application
+
+Most applications modelled by charms will also have a version - for example, the version of
+[Wordpress](https://charmhub.io/wordpress-k8s), [PostgreSQL](https://charmhub.io/postgresql-k8s), or
+[Jenkins](https://charmhub.io/jenkins).
+
+Each application will have its own versioning scheme, and
+its own way of accessing that information. To make things easier for admins, the charm can expose
+the workload version through Juju. You can find the workload (application) version with
+`juju status` in the application table in the "Version" column in the tabular view, or in the
+`applications.<app name>.version` field in JSON or YAML format. If the charm has not set the
+workload version, then the field will not be present in JSON or YAML format, and if the version
+string is too long or contains particular characters then it will not be displayed in the tabular
+format.
+
+To set the workload version, charms call
+[`Unit.set_workload_version()`](https://ops.readthedocs.io/en/latest/#ops.Unit.set_workload_version)
+providing the appropriate string (which is generally retrieved from the workload application
```

Comment:
We should find a systematic way to surface this sort of information in the charm SDK docs. Right now this shows up as a main piece of content only in the two tutorials. We should find a systematic way to expose it in the How-to guides. Not just this bit but every other bit in Ops. On the pattern of how we expose various bits of the CLI through How-to guides in the Juju docs.